code/modules/clothing/head/hardhat.dm

```diff
@@ -81,4 +81,92 @@
 	heat_protection = HEAD
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
 	cold_protection = HEAD
-	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
\ No newline at end of file
+	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
+
+/obj/item/clothing/head/hardhat/weldhat
+	name = "welding hard hat"
+	desc = "A piece of headgear used in dangerous working conditions to protect the head. Comes with a built-in flashlight AND welding shield! The bulb seems a little smaller though."
+	brightness_on = 3 //Needs a little bit of tradeoff
+	dog_fashion = null
+	actions_types = list(/datum/action/item_action/toggle_helmet_light, /datum/action/item_action/toggle)
+	flash_protect = 2
+	tint = 2
+	flags_inv = HIDEEYES
+	flags_cover = HEADCOVERSEYES
+	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT
+	visor_flags_inv = HIDEEYES 
+	visor_flags_cover = HEADCOVERSEYES
+
+/obj/item/clothing/head/hardhat/weldhat/Initialize()
+	..()
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/user)
+	toggle_helmet_light(user)
+	update_icon()

```

Comment:
You're already calling `update_icon()` at the end of `toggle_helmet_light()`, no need to call again.